src/sage/libs/gap/element.pyx

```diff
@@ -218,16 +218,41 @@ cdef Obj make_gap_integer(sage_int) except NULL:
 
     INPUT:
 
-    - ``sage_int`` -- Sage integer
+    - ``sage_int`` -- Sage integer or Python int
 
     OUTPUT: the integer as a GAP ``Obj``
 
     TESTS::
 
         sage: libgap(1)   # indirect doctest
         1
+        sage: libgap(int(10**30))  # indirect doctest
+        1000000000000000000000000000000
+        sage: libgap(-int(10**30))  # indirect doctest
+        -1000000000000000000000000000000
     """
-    return GAP_NewObjIntFromInt(<int>sage_int)
+    cdef mpz_t temp
+    cdef Obj result
+    cdef Int size
+    
+    # Try the fast path for small integers
+    try:
+        return GAP_NewObjIntFromInt(<int>sage_int)
+    except OverflowError:
+        # For large Python int, we need to convert via GMP
+        # and use GAP_MakeObjInt
+        mpz_init(temp)
+        try:
+            mpz_set_pylong(temp, sage_int)
+            size = mpz_size(temp)
+            if mpz_sgn(temp) < 0:
+                size = -size
+            # Access the limbs directly from the mpz_t structure
+            # temp[0]._mp_d gives us the pointer to the limbs
+            result = GAP_MakeObjInt(<const UInt*>temp[0]._mp_d, size)
+            return result
+        finally:
+            mpz_clear(temp)
```

Comment:
* does this really work with large **sage** integer?
* avoid blanket try/finally. in which situation would this block raise?
* how do you know a mpz limb has the same size as a gap limb?
* is there a non-internal function that access the limbs?